packages/embark/src/cmd/cmd_controller.js

```diff
@@ -408,13 +409,30 @@ class EmbarkController {
 
   }
 
-  async reset() {
-    const doneMessage = __("reset done!").green;
-    await embarkReset({doneMessage});
+  async reset(options) {
+    const embarkConfig = require(fs.dappPath(options.embarkConfig || 'embark.json'));
+
+    let removePaths = [];
+    let defaultPaths = [...defaultResetPaths];
+
+    defaultPaths.push(embarkConfig.buildDir);
+    embarkConfig.generationDir && defaultPaths.push(embarkConfig.generationDir);
```

Comment:
@jrainville @michaelsbradleyjr Notice that `generationDir` and `buildDir` are no part of defaults, ensuring changes to embark.json on that regard will be honored by embark-reset.